extensions/s3/build.gradle

```diff
@@ -13,6 +13,9 @@ dependencies {
     implementation project(':Configuration')
     implementation project(':log-factory')
 
+    // TODO Included this to get CleanupReferenceProcessorInstance, where should I move CleanupReferenceProcessorInstance?
+    implementation project(':engine-updategraph')
```

Comment:
What's this dependency for?

Reply:
Or we can split `CleanupReferenceProcessorInstance`. We really only need the liveness instance to live within engine-updategraph. The DEFAULT could live in the same module as `CleanupReferenceProcessor`.

Reply:
There's very little magic here. We could simplify to an enum that implements `Supplier<ReferenceQueue>` or similar, and we only need the reset functionality for the `LIVENESS` instance.